task9-1/hashTable.c

```diff
@@ -0,0 +1,93 @@
+#include "list.h"
+
+#define TABLE_SIZE 11
+
+typedef struct HashTable {
+    int size;
+    int filled;
+    List *table[TABLE_SIZE];
+} HashTable;
+
+HashTable *initTable() {
+    HashTable *table = malloc(sizeof(HashTable));
+    for (int i = 0; i < TABLE_SIZE; ++i) {
+        table->table[i] = initList();
+    }
+    table->filled = 0;
+    table->size = TABLE_SIZE;
+    return table;
+}
+
+int hash(char *key) {
+    int result = 0;
+    for (int i = 0; key[i] != '\0'; i++) {
+        result = (result + key[i]) % TABLE_SIZE;
+    }
+    return result;
+}
+
+void addToTable(HashTable *table, char *word) {
+    List *elementOfTable = (table->table[hash(word)]);
+    if(listLen(elementOfTable) == 0){
+        table->filled++;
+    }
+    add(elementOfTable, word);
+}
+
+void deleteTable(HashTable *table) {
+    for (int i = 0; i < table->size; i++) {
+        deleteList(table->table[i]);
+    }
+    free(table);
+}
+
+HashTable* readFileToTable(char* fileName) {
+    HashTable* table = initTable();
+    FILE* file = fopen(fileName, "r");
+    char word[CAPACITY];
+    while (fscanf_s(file, "%s ", word, CAPACITY) != EOF) {
+        addToTable(table, &word[0]);
```

Comment:
```suggestion
        addToTable(table, word);
```
Массивы автоматически кастаются к указателям